lang/py/src/avro/io.py

```diff
@@ -280,6 +292,61 @@ def read_utf8(self):
     """
     return unicode(self.read_bytes(), "utf-8")
 
+  def read_date_from_int(self):
+    """
+    Decode python date object from int
+    int stores the number of days from
+    the unix epoch, 1 January 1970 (ISO calendar).
+    """
+    days_since_epoc = self.read_int()
+    return datetime.date(1970, 1, 1) + datetime.timedelta(days_since_epoc)
+
+  def _build_time_object(self, value, unit_multiplier):
+    value = int(value * (1000000 // unit_multiplier))
```

Comment:
I don't think you need to have explicit typecasting here. Type casting is expensive so use it conservatively.